src/NetTopologySuite/Operation/OverlayNG/RobustClipEnvelopeComputer.cs

```diff
@@ -0,0 +1,104 @@
+﻿using NetTopologySuite.Geometries;
+
+namespace NetTopologySuite.Operation.OverlayNg
+{
+    /// <summary>
+    /// Computes a robust clipping envelope for a pair of polygonal geometries.
+    /// The envelope is computed to be large enough to include the full
+    /// length of all geometry line segments which intersect
+    /// a given target envelope.
+    /// This ensures that line segments which might intersect are
+    /// not perturbed when clipped using <see cref="RingClipper"/>.
+    /// </summary>
+    /// <author>Martin Davis</author>
+    internal class RobustClipEnvelopeComputer
```

Comment:
this isn't designed to be inherited from, so let's mark it `sealed`

(We can always unseal it later if that becomes a problem)